frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/TicfCriHandler.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.di.authentication.frontendapi.lambda;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import org.apache.logging.log4j.LogManager;
+import uk.gov.di.authentication.entity.TICFCRIRequest;
+import uk.gov.di.authentication.shared.services.CloudwatchMetricsService;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.SerializationService;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpTimeoutException;
+import java.time.Duration;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public class TicfCriHandler implements RequestHandler<TICFCRIRequest, Void> {
+
+    private final HttpClient httpClient;
+    private final ConfigurationService configurationService;
+    private final CloudwatchMetricsService cloudwatchMetricsService;
+
+    private static final SerializationService serialisationService =
+            SerializationService.getInstance();
+
+    public TicfCriHandler(
+            HttpClient httpClient,
+            ConfigurationService configurationService,
+            CloudwatchMetricsService cloudwatchMetricsService) {
+        this.httpClient = httpClient;
+        this.configurationService = configurationService;
+        this.cloudwatchMetricsService = cloudwatchMetricsService;
+    }
+
+    public TicfCriHandler(ConfigurationService configurationService) {
+        this.httpClient = HttpClient.newHttpClient();
+        this.configurationService = configurationService;
+        this.cloudwatchMetricsService = new CloudwatchMetricsService();
+    }
+
+    public TicfCriHandler() {
+        this(ConfigurationService.getInstance());
+    }
+
+    private static final org.apache.logging.log4j.Logger LOG =
+            LogManager.getLogger(TicfCriHandler.class);
+
+    @Override
+    public Void handleRequest(TICFCRIRequest input, Context context) {
+        LOG.info("received request to TICF CRI Handler");
```

Comment:
Is this going to be useful in production?  Better to assume its working and look for errors in prod so maybe make this a debug message that we only see in lower environments.

Reply:
switched to debug 👍